core/version.h

```diff
@@ -62,18 +62,26 @@
 // Example: 3.1.4 will be 0x030104, making comparison easy from script.
 #define VERSION_HEX 0x10000 * VERSION_MAJOR + 0x100 * VERSION_MINOR + VERSION_PATCH
 
+#ifdef REAL_T_IS_DOUBLE
+#define VERSION_DOUBLE ".double"
+#else
+// Single precision is the default, so there is no added suffix.
+#define VERSION_DOUBLE
+#endif // REAL_T_IS_DOUBLE
```

Comment:
`#define`s here are public so I'd prefer not to expose an empty define.

You could put the `#ifdef REAL_T_IS_DOUBLE` around the define of `VERSION_FULL_CONFIG` directly.